compiler/ballerina-lang/src/main/java/io/ballerina/projects/environment/PackageCache.java

```diff
@@ -34,31 +34,47 @@
 public interface PackageCache {
 
     /**
-     * Returns the package with the given {@code PackageId}.
-     *
+     * @deprecated 
+     * Use {@link #getPackage(PackageOrg, PackageName, PackageVersion)} instead.
+     * Returns the package with the given {@code PackageId}
+     * 
      * @param packageId the packageId
      * @return the package with the given {@code PackageId}.
      */
+    @Deprecated
     Optional<Package> getPackage(PackageId packageId);
 
     /**
+     * @deprecated
+     * Use {@link #getPackageOrThrow(PackageOrg, PackageName, PackageVersion)} instead.
      * Returns the package with the given {@code PackageId} or throw an {@code IllegalStateException}.
      *
      * @param packageId the packageId
      * @return the package with the given {@code PackageId} or throw an {@code IllegalStateException.
      */
+    @Deprecated
     Package getPackageOrThrow(PackageId packageId);
 
     /**
-     * Returns the package with the given organization, name and version.
+     * Returns the package with the given {@code PackageOrg}, {@code PackageName} and {@code PackageVersion}.
      *
      * @param packageOrg      organization name
      * @param packageName     package name
      * @param version package version
-     * @return the package with given organization, name and version
+     * @return the package with given {@code PackageOrg}, {@code PackageName} and {@code PackageVersion}
      */
     Optional<Package> getPackage(PackageOrg packageOrg, PackageName packageName, PackageVersion version);
 
+    /**
+     * Returns the package with the given {@code PackageOrg}, {@code PackageName} and {@code PackageVersion} or throw an {@code IllegalStateException}.
+     * 
+     * @param packageOrg
+     * @param packageName
+     * @param version
+     * @return the package with given {@code PackageOrg}, {@code PackageName} and {@code PackageVersion} or throw an {@code IllegalStateException}.
+     */
+    Package getPackageOrThrow(PackageOrg packageOrg, PackageName packageName, PackageVersion version);
```

Comment:
When adding new methods to the interface, the methods should be implemented in the classes that implement the interface.